src/binary/ibc-setup/commands/ics20.ts

```diff
@@ -0,0 +1,152 @@
+import { resolveKeyFileOption } from '../utils/options/shared/resolve-key-file-option';
+import { resolveMnemonicOption } from '../utils/options/shared/resolve-mnemonic-option';
+
+export type Flags = {
+  readonly interactive: boolean;
+  readonly home?: string;
+  readonly keyFile?: string;
+  readonly mnemonic?: string;
+  readonly src?: string;
+  readonly dest?: string;
+  readonly srcPort?: string;
+  readonly destPort: string;
+};
+
+export type Options = {
+  readonly home: string;
+  readonly mnemonic: string;
+  readonly src: string;
+  readonly dest: string;
+  readonly srcPort: string;
+  readonly destPort: string;
+};
+
+const defaultPort = 'transfer';
+
+export async function ics20(flags: Flags): Promise<void> {
+  const home = resolveHomeOption({ homeFlag: flags.home });
+  const app = loadAndValidateApp(home);
+
+  if (!app) {
+    throw new Error(`${appFile} not found at ${home}`);
+  }
+
+  const keyFile = resolveKeyFileOption({ keyFileFlag: flags.keyFile, app });
+  const mnemonic = await resolveMnemonicOption({
+    interactiveFlag: flags.interactive,
+    mnemonicFlag: flags.mnemonic,
+    keyFile,
+    app,
+  });
+  const src = resolveRequiredOption('src')(flags.src, process.env.RELAYER_SRC);
+  const dest = resolveRequiredOption('dest')(
+    flags.dest,
+    process.env.RELAYER_DEST
+  );
+  const srcPort = resolveRequiredOption('srcPort')(
+    flags.srcPort,
+    process.env.RELAYER_SRC_PORT,
+    defaultPort
+  );
+  const destPort = resolveRequiredOption('destPort')(
+    flags.destPort,
+    process.env.RELAYER_DEST_PORT,
+    defaultPort
+  );
+
+  run(
+    {
+      src,
+      dest,
+      home,
+      mnemonic,
+      srcPort,
+      destPort,
+    },
+    app
+  );
+}
+
+export async function run(options: Options, app: AppConfig): Promise<void> {
+  const registryFilePath = path.join(options.home, registryFile);
+  const { chains } = loadAndValidateRegistry(registryFilePath);
+  const srcChain = chains[options.src];
+  if (!srcChain) {
+    throw new Error('src chain not found in registry');
+  }
+  const destChain = chains[options.dest];
+  if (!destChain) {
+    throw new Error('dest chain not found in registry');
+  }
+  const ordering = Order.ORDER_UNORDERED;
+  const version = 'ics20-1';
+
+  const nodeA = await createClient(options.mnemonic, srcChain);
+  const nodeB = await createClient(options.mnemonic, destChain);
+  // TODO: Handle if connection flag is provided
+  const link = await Link.createWithNewConnections(nodeA, nodeB);
+  console.log(
+    `Created connections ${link.endA.connectionID} (${link.endA.clientID}) <=> ${link.endB.connectionID} (${link.endB.clientID})`
+  );
+
+  const srcClient = link.endA.clientID;
+  const destClient = link.endB.clientID;
+  const srcConnection = link.endA.connectionID;
+  const destConnection = link.endB.connectionID;
+  const appFilePath = path.join(options.home, appFile);
+  const appYaml = yaml.dump(
```

Comment:
I like how it just appends to what was there

Reply:
Not sure about one case:

**app.yaml**:
```
...
srcConnection: 'some-src-connection-1'
destConnection: 'some-dest-connection-1'
```

running `ibc-setup ics20 --srcConnection test-conn-1 --destConnection test-conn-1 ...`

will result in overriding `some-src-connection-1` and `some-dest-connection-1`. Is that the desired behavior?

Reply:
All channels must be on the same connections.
If we change it, the old ones will be invalid. If we ignore them, then the new channel cannot be made.

I bet this argument isn't good.

For the "advanced" manual channel creation commands, maybe we can allow input from app.yaml, flags, or env (assuming they don't edit app.yaml automatically). However, for this use case (ics20 should be easy to use), let us *only* take src/destConnection from `app.yaml`

----

TL;DR: just change the following


```ts
  const srcConnection = resolveOption(
    flags.srcConnection,
    process.env.RELAYER_SRC_CONNECTION,
    app.srcConnection
  );
  const destConnection = resolveOption(
    flags.destConnection,
    process.env.RELAYER_DEST_CONNECTION,
    app.destConnection
  );
```

to:

```ts
  const srcConnection = app.srcConnection;
  const destConnection = app.destConnection;
```

And unregister the flags and update the spec doc (maybe with an explanation). Then ready to merge

Reply:
48a9748
577b229